Better errors from codegen

Carter12s opened the pull request:
Closes: https://github.com/Carter12s/roslibrust/issues/126
Closes: https://github.com/Carter12s/roslibrust/issues/128

Trying to catch some common user errors and make them more clear. Investigated automated tests for these kinds of errors, but it looks like it is actually pretty annoying to setup. Looks like this will do the job, but a little much to setup right now: https://crates.io/crates/compiletest_rs

These test checks were manually performed by me while setting this up:
Manually modifying calling_services to introduce a typo on this line:
`roslibrust_codegen_macro::find_and_generate_ros_messages!("assets/ros1_common_interfacs",);`

Now produces the following compilation error when invoking `cargo run --example calling_service`
```
error: Failed to find any services or messages while generating ROS message definitions, paths searched: ["assets/ros1_common_interfacs"]
 --> roslibrust/examples/calling_service.rs:4:1
  |
4 | roslibrust_codegen_macro::find_and_generate_ros_messages!("assets/ros1_common_interfacs",);
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `roslibrust_codegen_macro::find_and_generate_ros_messages` (in Nightly builds, run with -Z macro-backtrace for more info)
```

ssnover reviewed (changes requested):
Rather than messages or services can you reference ROS packages? I think that could make it more clear that they can't be standalong message files on disk, they should be part of a package (which is what we search for).

Carter12s commented:
Continuing to add some better messages:

Invoking this line:
```rust
roslibrust_codegen_macro::find_and_generate_ros_messages!("assets/ros1_test_msgs");
```
Without std_msgs in ROS_PACKAGE_PATH now produces:

```
error: Unable to resolve dependencies after reaching search limit.
       The following messages have unresolved dependencies: ["test_msgs/Float64Stamped"]
       These messages likely depend on packages not found in the provided search paths.
 --> roslibrust/examples/basic_publisher.rs:6:1
  |
6 | roslibrust_codegen_macro::find_and_generate_ros_messages!("assets/ros1_test_msgs");
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `roslibrust_codegen_macro::find_and_generate_ros_messages` (in Nightly builds, run with -Z macro-backtrace for more info)
  ```
  
  
![image](https://github.com/Carter12s/roslibrust/assets/8097919/d742e36a-05be-4a50-96be-d7f14c442ca6)

Carter12s commented:
Alright @ssnover I think this one is really ready for review now.

I added a big'ol pile of error improvements, replaced all of the panic! and unwrap() I reasonably could with proper error bubbling, and got the whole thing to print much more understandable errors way more often!
